pyvista/core/filters.py

```diff
@@ -2265,23 +2266,52 @@ def extract_surface(dataset, pass_pointid=True, pass_cellid=True, inplace=False)
             Adds a cell array "vtkOriginalPointIds" that idenfities which
             original cells these surface cells correspond to
 
+        subdivision : int, optional
+            If the input is an unstructured grid with nonlinear faces,
+            this parameter determines how many times the face is
+            subdivided into linear faces.
+
+            If 0, the output is the equivalent of its linear
+            counterpart (and the midpoints determining the nonlinear
+            interpolation are discarded). If 1 (the default), the
+            nonlinear face is triangulated based on the midpoints. If
+            greater than 1, the triangulated pieces are recursively
+            subdivided to reach the desired subdivision. Setting the
+            value to greater than 1 may cause some point data to not
+            be passed even if no nonlinear faces exist. This option
+            has no effect if the input is not an unstructured grid.
+
         Returns
         -------
         extsurf : pyvista.PolyData
             Surface mesh of the grid
 
+        Examples
+        --------
+        Extract the surface of an UnstructuredGrid
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> grid = examples.load_hexbeam()
+        >>> surf = grid.extract_surface()
+        >>> type(surf)
+        <class 'pyvista.core.pointset.PolyData'>
+
         """
         surf_filter = _vtk.vtkDataSetSurfaceFilter()
         surf_filter.SetInputData(dataset)
         if pass_pointid:
             surf_filter.PassThroughCellIdsOn()
         if pass_cellid:
             surf_filter.PassThroughPointIdsOn()
-        surf_filter.Update()
 
-        # TODO: Add
-        # surf_filter.SetNonlinearSubdivisionLevel(subdivision)
+        if subdivision != 1:
+            surf_filter.SetNonlinearSubdivisionLevel(subdivision)
```

Comment:
I guess the VTK name makes the functionality very clear. It would probably be more obvious if we also called the parameter `nonlinear_subdivision`, but that might be impractically unwieldy.

Reply:
Zen of Python tells me that this is a balance between:
```
Explicit is better than implicit.
Simple is better than complex.
```
However, considering that those not well versed in quadratic elements might not understand how you can "subdivide" the surface of an element, or even why you' want to do it here, I think you're right.  I'll implement it.

Reply:
Yeah, I'm on the fence myself.